File: Tarea 4/Factorial.py
import os

def archivoTXT(N, resultado):
	try:
		archivo = open("HistorialDeFactoriales.txt", "a")
		archivo.write("El factorial de " + str(N) + " es " + str(resultado) + "\n")
		archivo.close()
	except:
		print ("No se encontro el archivo .txt :(")

def validarNumero(mensaje, advertencia):
	while True:
		try:
			N = int(input(mensaje))
			if (N >= 0):
				return N 
				break
			else:
				print (advertencia + "\n")
		except ValueError:
			print (advertencia + "\n")

def nFactorial(N):
	resultado = 1
	for i in range (N):
		resultado = (i+1)*resultado
	print ("El factorial de " + str(N) + " es " + str(resultado) + "\n")
	archivoTXT(N, resultado)

def calculadoraFactorial():
	while True:
		os.system("cls")
		print ("--- Calculadora factorial de N! ---" + "\n")
		N = validarNumero("Ingrese el numero deseado: ", "¡Entrada no valida!")
		nFactorial(N)
		delay = input("Presione cualquier tecla para continuar (s para salir) . . . ")
		os.system("cls")
		if (delay == "s" or delay == "S"):
			break

calculadoraFactorial()

File: Tarea 4/Factorial.m
function calculadoraFactorial()
  while(true)
    clc
    disp(strcat("--- Calculadora factorial de N! ---", "\n"))
    N = validarNumero("Ingrese el numero deseado: ", "¡Entrada no valida!");
    nFactorial(N)
    delay = input("Presione cualquier tecla para continuar (s para salir) . . . ", 's');
    clc
    if(delay == "s" || delay == "S")
      break
    endif
  endwhile
endfunction

function archivoTXT(N, resultado)
  try
    archivo = fopen('HistorialDeFactoriales.txt', 'a');
    fprintf(archivo, "El factorial de %d es %d \n", N, resultado);
    fclose(archivo);
  catch
    disp("No se encontro el archivo .txt :(")
  end_try_catch
endfunction

function N = validarNumero(mensaje, advertencia)
  while(true)
    try
      N = input(mensaje);
      if(N >= 0)
        break
      else
        disp(strcat(advertencia, "\n"))
      endif
    catch
      disp(strcat(advertencia, "\n"))
    end_try_catch
  endwhile
endfunction

function nFactorial(N)
  resultado = 1;
  for i = 0 : (N - 1)
    resultado = (i+1)*resultado;
  endfor
  fprintf("El factorial de %d es de %d \n\n", N, resultado)
  archivoTXT(N, resultado);
endfunction